CorsixTH/Lua/room.lua

```diff
@@ -493,19 +493,37 @@ function Room:commandEnteringStaff(humanoid, already_initialized)
   end
 end
 
+
+
+--! Activates and deactivates the staff waiting for patient mood icon
+-- and dynamic info text
+--!param activate (bool) - true to activate, false or nil to deactivate
+function Room:_staffWaitToggle(activate)
+  local state = "deactivate"
+  local dynamic_text = ""
+
+  if activate then
+    dynamic_text = _S.dynamic_info.staff.actions.waiting_for_patient
+    state = "activate"
+  end
+
+  if not self.staff_member_set and self.staff_member then
+    self.staff_member:setMood("staff_wait", state)
+    self.staff_member:setDynamicInfoText(dynamic_text)
+  else
+    for staff_member in pairs(self.staff_member_set) do
+      staff_member:setMood("staff_wait", state)
+      staff_member:setDynamicInfoText(dynamic_text)
```

Comment:
`staff_member:setDynamicInfoText(dynamic_text)` will have a clash with #2125.
Did you want this PR to go first, or to wait for the other PR?